daemon/cmd/kube_proxy_replacement.go

```diff
@@ -362,20 +374,20 @@ func initKubeProxyReplacementOptions() (strict bool) {
 		// running in full KPR mode as otherwise conntrack would be
 		// required for NAT operations
 		if !option.Config.KubeProxyReplacementFullyEnabled() {
-			log.Fatalf("%s requires the agent to run with %s=%s.",
+			return false, fmt.Errorf("%s requires the agent to run with %s=%s.",
 				option.InstallNoConntrackIptRules, option.KubeProxyReplacement, option.KubeProxyReplacementStrict)
 		}
 
 		if !option.Config.EnableBPFMasquerade {
-			log.Fatalf("%s requires the agent to run with %s.",
+			return false, fmt.Errorf("%s requires the agent to run with %s.",
```

Comment:
nit, this function has a couple of naked `return` statements which it looks like later in the commit the other functions do an explicit `return nil`. Should probably make it consistent. In general we tend to avoid named return vars in any functions other than the shortest, simplest functions, just to avoid unintentionally returning some value that's in the variable at some point during the function.

Reply:
agreed